controlplane/pkg/apis/nsmdapi/nsmd.proto

```diff
@@ -27,7 +28,15 @@ message DeleteConnectionReply {
 
 }
 
+message EnumConnectionRequest {
+}
+
+message EnumConnectionReply {
+    repeated string workspace = 1;
+}
+
 service NSMD {
     rpc RequestClientConnection (ClientConnectionRequest) returns (ClientConnectionReply);
+    rpc EnumConnection (EnumConnectionRequest) returns (EnumConnectionReply);
```

Comment:
We might want to use a prefix like "List" instead of "Enum" (might get confused with actual enums :) ).